lib/mayaHydra/mayaPlugin/renderOverride.cpp

```diff
@@ -1864,7 +1898,12 @@ bool MtohRenderOverride::select(
         }
     }
 
-    _PopulateSelectionList(outHits, selectInfo, selectionList, worldSpaceHitPts);
+    //isOneNodeInComponentsPickingMode will be true if one of the picked node is in components picking mode
+    bool isOneNodeInComponentsPickingMode = false;
+    _PopulateSelectionList(outHits, selectInfo, selectionList, worldSpaceHitPts, isOneNodeInComponentsPickingMode);
+    if (isOneNodeInComponentsPickingMode){
+        return false;//When being in components picking on a node, returning false will use maya/OGS for components selection
+    }
```

Comment:
Question : is component-picking not simply handled by the `if (isInComponentsPickingMode(selectInfo))` check earlier? I think I remember a mention that there were different cases of component picking but I can't remember the other ones (other than right-click a selected object and go into component selection mode)

Reply:
You're right that this needs clarification, so I've added the following comments to the code :
/*
* There are 2 modes of selection picking for components in maya :
* 1) You can be in components picking mode, this setting is global.This is detected in the function "isInComponentsPickingMode(selectInfo)"
* 2) The second mode is when you right click on a node and choose a component to pick it (e.g : Face), this is
* where we use the variable "isOneNodeInComponentsPickingMode" to detect that case, later in this function.
*/